lib/sneakers/queue.rb

```diff
@@ -59,6 +59,8 @@ def unsubscribe
     # XXX can we cancel bunny and channel too?
     @consumer.cancel if @consumer
     @consumer = nil
+  rescue Bunny::ConnectionClosedError
+    sleep @bunny.instance_variable_get(:@network_recovery_interval)
```

Comment:
Sure, I can do that.  The sleep here was merely meant to stop Sneakers from thrashing but isn't entirely necessary.  I can remove it for the time being.